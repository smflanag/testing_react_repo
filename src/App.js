import React from 'react';
import './App.css';
import Season from './season';
import SeasonList from './seasonlist';
import SeasonDetail from './seasondetail';
import {
  Route,
  Switch,
  BrowserRouter,
  Redirect,
  NavLink,
} from "react-router-dom";

function App() {
  return (
    <div className="App">
        <p>
          This is a single page app test to learn testing, and accessibility.
        </p>

      <div>
          <BrowserRouter>
              <ul className="list">
                 <li><NavLink to="/seasonlist">Season List</NavLink></li>
              </ul>


            <Switch>
              <Route exact path="/seasonlist" component={SeasonList} />
              <Route exact path={`/seasons/:seasonId`} component={SeasonDetail}/>
            </Switch>
            </BrowserRouter>
        </div>
    </div>
  );
}

export default App;
