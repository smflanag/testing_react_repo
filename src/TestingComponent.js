import React from 'react';

function TestingComponent() {
  return (
    <div className="testingcomponent">
      <header className="App-header">
        <p>
          This is the test component.
        </p>
        <a
          className="App-link"
          href="https://reactjs.org"
          target="_blank"
          rel="noopener noreferrer"
        >
          Learn React
        </a>
      </header>
    </div>
  );
}

export default TestingComponent;