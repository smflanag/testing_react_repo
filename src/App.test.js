import React from 'react';
import ReactDOM from 'react-dom';
import App from './App';
import { create } from "react-test-renderer";
import TestingComponent from "./TestingComponent";

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<App />, div);
  ReactDOM.unmountComponentAtNode(div);
});

const sum = require("./sum");

test('test that sum(2,1) equals 3', () => {
    expect(sum(2,1)).toBe(3);
});



test("testing component matches the snapshot", () => {
const component = create(<TestingComponent />);
expect(component.toJSON()).toMatchSnapshot();
});
