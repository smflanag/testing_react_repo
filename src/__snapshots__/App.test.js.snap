// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing component matches the snapshot 1`] = `
<div
  className="testingcomponent"
>
  <header
    className="App-header"
  >
    <p>
      This is the test component.
    </p>
    <a
      className="App-link"
      href="https://reactjs.org"
      rel="noopener noreferrer"
      target="_blank"
    >
      Learn React
    </a>
  </header>
</div>
`;
